packages/client/src/components/SignIn/index.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState }  from 'react';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Text from '../../components/TextBox';
+import Checkbox from '@material-ui/core/Checkbox';
+import useAuth from '../../providers/AuthProvider'
+import Button from '../Button';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import IconButton from '@material-ui/core/IconButton';
+import { Visibility, VisibilityOff } from '@material-ui/icons';
+import styled from 'styled-components';
+import Notification from '../Notification'
+
+const AuthInner = styled.div`
+    width: 450px;
+    margin: auto;
+    background: #ffffff;
+    box-shadow: 0 14px 80px rgba(34, 35, 58, 0.2);
+    padding: 40px 55px 45px 55px;
+    border-radius: 15px;
+    transition: all .3s;
+`
+const AuthWrapper = styled.div`
+    display: flex;
+    justify-content: center;
+    flex-direction: column;
+    text-align: left;
+    
+    .form-control:focus {
+        border-color: #00568C;
+        box-shadow: none;
+    }
+    h3 {
+        text-align: center;
+        margin: 0;
+        line-height: 1;
+        padding-bottom: 20px;
+    }
+`
+const PassForget = styled.p`
+    text-align: right;
+    font-size: 13px;
+    padding-top: 10px;
+    color: #7f7d7d;
+    margin: 0;
+    
+    a {
+        color: #00568C;
+    }
+`
+
+const handleMouseDownPassword = (event) => {
+    event.preventDefault();
+};
+
+
+const Signin = () => {
+    const { signin } = useAuth();
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    let error = false;
+    const [showPassword, setShowPassword] = useState(false);
+    const [serverError, setServerError] = useState(false);
+    const [remember, setRemember] = useState(false);
+
+    const handleChange = (prop) => {
+        if (prop === 'remember') {
+            setRemember(!remember);
+        } else if (prop === 'serverError') {
+            setServerError(!serverError);
+        } else if (prop === 'email') {
+            setEmail(!email);
+        } else if (prop === 'password') {
+            setPassword(!password);
+        } else if (prop === 'error') {
+            error = !error;
+        }
+    }
+    const handleClickShowPassword = () => {
+        setShowPassword(!showPassword);
+    };
+    // TODO: fix validation
+    const checkValidation = () => {
+        error = !(email.length > 0 && email.includes('@') && email.includes('.') && password.length > 5);
+    }
+
+    return (
+        <AuthWrapper>
+            <AuthInner>
+                <form>
+                    <h1>Sign In</h1>
+                    <div className="form-group">
+                        <Text
+                            autoFocus = {true}
+                            id = 'email'
+                            placeholder = "Email"
+                            value = {email}
+                            onChange = {handleChange('email')}
```

Comment:
you want `() => handleChange('email')` here, currently this will actually call `handleChange` every render which sets state and triggers rerender

Reply:
So that is the one I missed. Thanks!